src/endpoint.cpp

```diff
@@ -414,6 +416,10 @@ void Endpoint::_add_sys_comp_id(uint8_t sysid, uint8_t compid)
     if (has_sys_comp_id(sys_comp_id)) {
         return;
     }
+    if ((sniffer_sysid != 0) && ((sys_comp_id >> 8) == sniffer_sysid))
```

Comment:
```suggestion
    if (sniffer_sysid != 0 && sysid == sniffer_sysid)) {
```

Humn... I noticed you run clang-format, but not sure why it left he if without braces here pass. @jbeyerstedt ?